lib/puppet/ssl/host.rb

```diff
@@ -395,43 +395,38 @@ def state
 
   private
 
-  # Ensures that CRL is either on disk, or that CRL checking has been disabled
+  # Ensures that CRL is on disk; downloads the CRL from server if not found
+  # Does nothing if use_crl? is set not truthy
   def ensure_crl_if_needed
     if use_crl? && !Puppet::FileSystem.exist?(crl_path)
-      # The CertificateRevocationList indirector will attempt to download the
-      # CRL from the CA if it does not exist on disk. It will ask host for
-      # its ssl_store again, but expect crl checking to be disabled for that
-      # store. This is not thread safe, and should be replaced as soon as we
-      # no longer need to use the indirector to download the CRL (PUP-8654).
-      old_crl_usage = @crl_usage
-      @crl_usage = false
-      Puppet.debug _("Disabling certificate revocation checking when fetching the CRL and no CRL is present")
-      if !CertificateRevocationList.indirection.find(CA_NAME)
-        raise Puppet::Error,
-              _("Certificate revocation checking is enabled but a CRL cannot be found; CRL checking will not be performed.")
-      end
-      @crl_usage = old_crl_usage
+      download_and_save_crl_bundle
+      load_crls(crl_path)
```

Comment:
So my concern (which we talked about) was that after the file is written, we check the contents (load_crls), but if an error is raised there, we don't delete the file.  So... !Puppet::FileSystem.exist?(crl_path) would be FALSE going forward and we wouldnt try to get a new crl.  Maybe we could write up a ticket to address that and expected behavior?

Reply:
My other thought is that maybe load_crls should not be called here.  Its already called in puppet/lib/puppet/ssl/host.rb:625 in order to get the crls to add them to the store.  Here we get the crls, but we don't use them.  So seems a bit redundant.  Thoughts?

Reply:
I think we should have a ticket about the first issue you described; we may want to err on the side of leaving behind data for people to inspect when errors happen, or if someone is manually changing their data, we might not want to be deleting files they have placed there. We may also want to add something to the command-line tool to deal with files we can't seemingly load.

I'm fine removing the `load_crls` in `ensure_crls_if_needed` method; it's a little overly cautious. 🎸

Reply:
We might want to use https://tickets.puppetlabs.com/browse/SERVER-2207 as that ticket.